### PR TITLE
Learning module dead code

### PR DESCRIPTION
**Pull request by islemaster**
# Description

While researching the mysteries of the PLC data model, Ha and I found some unreachable / useless code that can be safely refactored away.

Hoping we get to tear out a _lot_ more of this in the near future, but removing annoyances as we find them.

## Testing story

The code being refactored already has test coverage... and obviously did not have coverage for the unreachable cases being removed.

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [x] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: dashboard/app/controllers/plc/enrollment_evaluations_controller.rb [hunk 1]
```diff
@@ -19,7 +19,7 @@ def confirm_assignments
 
     modules_to_enroll_in = Plc::LearningModule.find([params[:content_module], params[:practice_module]])
 
-    if modules_to_enroll_in.size == 2 && modules_to_enroll_in.pluck(:module_type).sort == [Plc::LearningModule::CONTENT_MODULE, Plc::LearningModule::PRACTICE_MODULE]
+    if modules_to_enroll_in.pluck(:module_type).sort == [Plc::LearningModule::CONTENT_MODULE, Plc::LearningModule::PRACTICE_MODULE]
```
Comment (islemaster) [inline dashboard/app/controllers/plc/enrollment_evaluations_controller.rb:22]: The second part of the conditional implicitly checks that there are two elements, and I can't see any performance reason to shortcut this check. (We're running the query a couple of lines above either way.)

Also, TIL [`find` can be used to retrieve multiple objects.](https://guides.rubyonrails.org/active_record_querying.html#find)

File: dashboard/app/models/plc/enrollment_unit_assignment.rb [hunk 2]
```diff
@@ -76,7 +76,7 @@ def summarize_progress
     # If the course unit has an evaluation level, then status is determined by the completion of the focus group modules
     if plc_course_unit.has_evaluation?
       Plc::LearningModule::MODULE_TYPES.select {|type| categories_for_stage.include?(type)}.each do |flex_category|
-        module_category = flex_category || Plc::LearningModule::CONTENT_MODULE
+        module_category = flex_category
```
Comment (islemaster) [inline dashboard/app/models/plc/enrollment_unit_assignment.rb:79]: There are no falsy values in the `MODULE_TYPES` constant. Therefore, we will never fall back on this or-case.

File: dashboard/app/controllers/plc/enrollment_evaluations_controller.rb (hunk 1 above)
Comment (Hamms) [reply dashboard/app/controllers/plc/enrollment_evaluations_controller.rb:22]: Oh, wow. I would have assumed that `.find([a, b])` is closer to `.find(a) || .find(b)` than `[.find(a), .find(b)]`

File: dashboard/app/models/plc/enrollment_unit_assignment.rb [hunk 3]
```diff
@@ -76,7 +76,7 @@ def summarize_progress
...
         category_name = I18n.t("flex_category.#{module_category}")
```
Comment (Hamms) [inline dashboard/app/models/plc/enrollment_unit_assignment.rb:80]: should this have a default?

**Review by Hamms — APPROVED** [review]
Wooo cleanup!

File: dashboard/app/models/plc/enrollment_unit_assignment.rb (hunk 3 above)
Comment (islemaster) [reply dashboard/app/models/plc/enrollment_unit_assignment.rb:80]: Nice catch - I'll do a follow-up. [PLC-501](https://codedotorg.atlassian.net/browse/PLC-501)

**Pull request merged**